src/connectors/infinite-hits/__tests__/connectInfiniteHits-test.ts

```diff
@@ -596,23 +591,121 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/infinite-hi
       helper,
     });
 
-    expect(rendering).toHaveBeenCalledTimes(4);
-    expect(rendering).toHaveBeenLastCalledWith(
+    expect(renderFn).toHaveBeenCalledTimes(4);
+    expect(renderFn).toHaveBeenLastCalledWith(
       expect.objectContaining({
         hits: [{ objectID: 'a' }, { objectID: 'b' }],
       }),
       false
     );
   });
 
+  describe('dispose', () => {
+    it('calls the unmount function', () => {
+      const helper = algoliasearchHelper({} as Client, '', {});
+
+      const renderFn = (): void => {};
+      const unmountFn = jest.fn();
+      const makeWidget = connectInfiniteHits(renderFn, unmountFn);
+      const widget = makeWidget({});
+
+      expect(unmountFn).toHaveBeenCalledTimes(0);
+
+      widget.dispose!({ helper, state: helper.state });
+
+      expect(unmountFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('does not throw without the unmount function', () => {
+      const helper = algoliasearchHelper({} as Client, '', {});
+
+      const renderFn = (): void => {};
+      const makeWidget = connectInfiniteHits(renderFn);
+      const widget = makeWidget({});
+
+      expect(() =>
+        widget.dispose!({ helper, state: helper.state })
+      ).not.toThrow();
+    });
+
+    it('removes the TAG_PLACEHOLDER from the `SearchParameters`', () => {
+      const helper = algoliasearchHelper({} as Client, '', {
+        ...TAG_PLACEHOLDER,
+      });
+
+      const renderFn = (): void => {};
+      const makeWidget = connectInfiniteHits(renderFn);
+      const widget = makeWidget({});
+
+      expect(helper.state.highlightPreTag).toBe(
+        TAG_PLACEHOLDER.highlightPreTag
+      );
+
+      expect(helper.state.highlightPostTag).toBe(
+        TAG_PLACEHOLDER.highlightPostTag
+      );
+
+      const nextState = widget.dispose!({
+        helper,
+        state: helper.state,
+      }) as SearchParameters;
+
+      expect(nextState.highlightPreTag).toBeUndefined();
+      expect(nextState.highlightPostTag).toBeUndefined();
+    });
+
+    it('removes the TAG_PLACEHOLDER from the `SearchParameters` only with `escapeHTML`', () => {
```

Comment:
```suggestion
    it('does not remove the TAG_PLACEHOLDER from the `SearchParameters` with `escapeHTML`', () => {
```

I had trouble understanding the test because of the name.